### ISSUE TITLE
train_t5xxl=True, causes an error in sd-scripts

### ISSUE THREAD

**Issue by brundlesprout** [Kaoru8/T5XXL-Unchained#5]
where do you set the flag? I placed it in the scripts file, but it "returned non-zero exit status 2." i'm working off of the current SD3 branch.

accelerate launch --mixed_precision bf16 --num_cpu_threads_per_process 1 flux_train.py `
.
.
   --model_prediction_type raw `
   --guidance_scale 1.0 `
   --fused_backward_pass `
   --blocks_to_swap 10 `
   --train_batch_size 1 `
   --single_blocks_to_swap 0 `
   --full_bf16 `
   --apply_t5_attn_mask `
   --train_t5xxl

thank you

**Comment by brundlesprout** [comment]
I forgot to state that that i'm trying to finetune the base model, not build a lora.

**Comment by Kaoru8** [comment]
If you still need help with this - what does the actual traceback/error say? There's no information to go off of with just the command you used to launch training...

**Comment by brundlesprout** [comment]
Thanks - It's' been a while, so if i remember correctly, i was referring to the train_t5xxl flag. When i tried to set that in the accelerate launch, kohya scripts, command, the tool errored. **This is referencing your item 6. in the readme**, second bullet point:

**- Make sure that you're training both the UNet and T5, so use the train_t5xxl=True flag. Training just the UNet is of no use - both models need to be trained on the new tokens and embedding size in tandem to learn and adapt to them.**


I was trying to do a **full finetune**, not a LORA, so i think the issue is that that flag does not exist in the "flux_train", but only in the "flux_train_network" module.


running the accelerate launch command bellow will initiate this error
**flux_train.py: error: unrecognized arguments: --train_t5xxl=True** 
------------------
accelerate launch --mixed_precision bf16 --num_cpu_threads_per_process 1 flux_train.py `
   --pretrained_model_name_or_path "./models/unet/flux1-dev.safetensors" `
   --clip_l "./clip/clip_l.safetensors" `
   --t5xxl "./clip/t5xxl_fp16.safetensors" `
   --ae "./models/vae/ae.safetensors" `
   --save_model_as safetensors `
   --sdpa --persistent_data_loader_workers --max_data_loader_n_workers 2 `
   --seed 42 `
   --gradient_checkpointing `
   --mixed_precision bf16 `
   --save_precision bf16 `
   --config_file "./Kohya1.toml" `
   --learning_rate 5e-5 `
   --max_train_epochs 60  `
   --save_every_n_epochs 10 `
   --sdpa --highvram `
   --cache_text_encoder_outputs_to_disk `
   --cache_latents_to_disk `
   --save_every_n_epochs 1 `
   --optimizer_type adafactor `
   --optimizer_args "" `
   --lr_scheduler constant_with_warmup `
   --max_grad_norm 0.0 `
   --timestep_sampling shift `
   --discrete_flow_shift 3.1582 `
   --model_prediction_type raw `
   --guidance_scale 1.0 `
   --fused_backward_pass `
   --blocks_to_swap 10 `
   --full_bf16 `
   -**-train_t5xxl `**


Thank you.

**Comment by Kaoru8** [comment]
I think I know what the problem is... Try using `--network_args "train_t5xxl=True"` instead of just `--train_t5xxl=True`.

**Comment by brundlesprout** [comment]
thank you. I asked kohya about this, and his answer was: 

**Fine tuning is only available for FLUX.1 models. Text Encoder cannot be trained at the same time.
If you want to train Text Encoder, please do so separately using the transformers library, etc.**

I guess what he is saying is that the text encoder can't be trained at the same time when finetuning flux?   It's all a bit confusing. If I understand correctly when you are training the text encoders you are not actually altering the CLIP/T5, but rather the model's internal text projection blocks...so i assume that those blocks can't be trained at the same time when doing flux finetuning?

Any case, thanks for your replies. Finetuning with the kohya/comfyui modified code in this git does work, even w/out setting the T5 training flag, but off course, that means then the internal projection blocks are not trained when "finetuning" flux...I'll have to try a LORA. 

i did some initial testing with flux1Dev2pro_v10, but i believe that model to be overbaked. if anyone reads this, my recomm. is to start with the Flux-dev base model.

**Comment by Kaoru8** [comment]
Yeah, that's a confusing answer... Search the Kohya repo's readme for `CLIP-L and T5XXL LoRA Support`, and it's definitely supported. But yeah, you're not actually training the T5 model itself, you're just baking in adjustments to T5's weights into the LORA.

Glad you got it working, but heads up - plenty of people have issues training with the Flux1-Dev base models because of guidance distillation, which is why Flux-Dev2Pro was made in the first place. Yes, it's overbaked and IMO has inferior outputs during inference, but I've had better luck with it for training. So basically, use Flux-Dev2Pro to actually train the LORA (`pretrained_model_name_or_path` parameter), then use the resulting LORA with any base Flux1-Dev model for better inference results.

**Comment by brundlesprout** [comment]
Thanks again! I'll have to try again the FLux-Dev2Pro, but build a LoRA, not finetune.

**Comment by Dervlex** [comment]
@brundlesprout how is it going ? Any goal/success reached ?
